PURL-TYPES.rst

```diff
@@ -255,6 +255,37 @@ rpm
       pkg:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
       pkg:rpm/centerim@4.22.10-1.el6?arch=i686&epoch=1&distro=fedora-25
 
+swift
+----
+``swift`` for Swift packages:
+
+- There is currently no default repository for Swift packages.
+- The ``namespace`` is used for the scope of a Swift package.
+  It is not case sensitive and must be lowercased.
+
+  - A package scope consists of alphanumeric characters and hyphens.
+    Hyphens may not occur at the beginning or end,
+    nor consecutively within a scope.
+  - The maximum length of a package scope is 39 characters.
+  - A valid package scope matches the following regular expression pattern
+    ``\A[a-zA-Z0-9](?:[a-zA-Z0-9]|-(?=[a-zA-Z0-9])){0,38}\z``
+
+- The ``name`` is the name declared in the Swift package manifest.
+  It is case and normalization insensitive,
+  and must be normalized using Unicode Normalization Form KC (NFKC)
+  and then transformed with locale-independent case folding.
+
+  - The maximum length of a package name is 128 characters.
+  - A valid package name matches the following regular expression pattern
+    ``\A\p{XID_Start}\p{XID_Continue}{0,127}\z``
+
+- The ``version`` is the version of the package release,
+  and must be a valid SemVer version number.
+- Examples::
+
+      pkg:swift/alamofire/Alamofire@5.4.3
```

Comment:
Since there's no Swift default package registry just using namespace, name, & version seems like it could be ambiguous. If we had this URL:
```
github.com/alamofire/Alamofire
```
Then Alamofire decided to migrated to gitlab:
```
gitlab.com/alamofire/Alamofire    (not real, I made it up)
```

Maybe they keep their old (now stale) repo on github, but newer versions only go to gitlab.
Which repo does `pkg:swift/alamofire/Alamofire@5.4.3` refer to?

I'm on at Sonatype uses purl internally, and, to avoid ambiguity, we put the hostname in there. Tweaking the above example:
```
pkg:swift/github.com/alamofire/Alamofire@5.4.3
```

That would resolve the ambiguity.

Alternatively we could tack on the repository_url qualifier, but that'd require us defining a default repository for Swift. Which we could do but what would it be? github?

Reply:
I also found out [OSS Index does the same thing](https://ossindex.sonatype.org/ecosystem/swift) with their Swift purl strings. Their example:

```
pkg:swift/github.com/Alamofire/Alamofire
```

OSS Index is hosted and maintained by a different Sonatype team than the one I represent, but if two teams came to the same conclusion perhaps including the host name is a good idea.

Reply:
[This competing PR](https://github.com/package-url/purl-spec/pull/116) fits with the comments in this thread, and with what OSS Index does.

full disclosure: the author of that PR and I are on the same team at Sonatype.